server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryMailDAO.java

```diff
@@ -142,23 +141,23 @@ private PreparedStatement prepareSelect(Session session) {
     }
 
     @Override
-    public CompletableFuture<Void> store(MailRepositoryUrl url, Mail mail, BlobId headerId, BlobId bodyId) throws MessagingException {
-        return executor.executeVoid(insertMail.bind()
-            .setString(REPOSITORY_NAME, url.asString())
-            .setString(MAIL_KEY, mail.getName())
-            .setString(HEADER_BLOB_ID, headerId.asString())
-            .setString(BODY_BLOB_ID, bodyId.asString())
-            .setString(STATE, mail.getState())
-            .setString(SENDER, mail.getMaybeSender().asString(null))
-            .setList(RECIPIENTS, asStringList(mail.getRecipients()))
-            .setString(ERROR_MESSAGE, mail.getErrorMessage())
-            .setString(REMOTE_ADDR, mail.getRemoteAddr())
-            .setString(REMOTE_HOST, mail.getRemoteHost())
-            .setLong(MESSAGE_SIZE, mail.getMessageSize())
-            .setTimestamp(LAST_UPDATED, mail.getLastUpdated())
-            .setMap(ATTRIBUTES, toRawAttributeMap(mail))
-            .setMap(PER_RECIPIENT_SPECIFIC_HEADERS, toHeaderMap(mail.getPerRecipientSpecificHeaders()))
-        );
+    public Mono<Void> store(MailRepositoryUrl url, Mail mail, BlobId headerId, BlobId bodyId) {
+        return Mono.fromCallable(() -> insertMail.bind()
+                .setString(REPOSITORY_NAME, url.asString())
+                .setString(MAIL_KEY, mail.getName())
+                .setString(HEADER_BLOB_ID, headerId.asString())
+                .setString(BODY_BLOB_ID, bodyId.asString())
+                .setString(STATE, mail.getState())
+                .setString(SENDER, mail.getMaybeSender().asString(null))
+                .setList(RECIPIENTS, asStringList(mail.getRecipients()))
+                .setString(ERROR_MESSAGE, mail.getErrorMessage())
+                .setString(REMOTE_ADDR, mail.getRemoteAddr())
+                .setString(REMOTE_HOST, mail.getRemoteHost())
+                .setLong(MESSAGE_SIZE, mail.getMessageSize())
+                .setTimestamp(LAST_UPDATED, mail.getLastUpdated())
+                .setMap(ATTRIBUTES, toRawAttributeMap(mail))
+                .setMap(PER_RECIPIENT_SPECIFIC_HEADERS, toHeaderMap(mail.getPerRecipientSpecificHeaders())))
+            .flatMap(executor::executeVoidReactor);
```

Comment:
Is there a reason to not directly call `executor.executeVoidReactor`?

Reply:
yes, `mail.getMessageSize()` can throw, so we want an exception here to be wrapped as a Mono.error.

Reply:
ok, thanks

Reply:
then It's worth to have `CassandraAsynExecutor.executeVoidReactor(Callable<Statement>)` method?

Reply:
I would say no